solutions/cf-20C.mdx

```diff
@@ -0,0 +1,78 @@
+---
+id: cf-20C
+source: CF
+title: Dijkstra?
+author: Jesse Choe
+---
+
+We can implement Dijkstra's Shortest Path algorithm to find the shortest distances between node $1$ and node $n$. To find the path that yields the shortest path between nodes $1$ and $n$, we should backtrack from node $n$ and determine if a given node should be in the path using the shortest distance array. However, when the shortest distance from node 1 to node $n$ is $\infty$, then we should print $-1$. 
+
+**Time Complexity:** $\mathcal{O}(N\log M)$
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+
+using namespace std;
+using ll = long long;
+using vi = vector<int>;
+using pi = pair<ll, ll>;
+
+#define f first
+#define s second
+#define all(x) begin(x), end(x)
+
+vector<pi> adj[100000];
+ 
+const ll INF = 1e11;
+	
+int main() {
+	priority_queue<pi, vector<pi>, greater<pi>> pq;
+	int n, m; cin >> n >> m;
+	for(int i = 0; i < m; i++){
+		int a, b, w; cin >> a >> b >> w;
+		a--, b--; adj[a].push_back({w, b}), adj[b].push_back({w, a});
+	}
+	vector<ll> dist(n, INF); dist[0] = 0;
+	vector<bool> visited(n, false);
+	pq.push({0, 0});
+	while(!pq.empty()){
+		pi a = pq.top();
+		pq.pop();
+		if(visited[a.s]) continue;
+		visited[a.s] = true;
+		for(pi vertices: adj[a.s]){
+			if(dist[a.s] + vertices.f < dist[vertices.s]){
+				dist[vertices.s] = dist[a.s] + vertices.f;
+				pq.push({dist[vertices.s], vertices.s});
+			}
+		}
+	}
+	if(dist[n - 1] == INF){
+		cout << -1 << endl;
+	} else {
```

Comment:
why spaces around the `else` but not around any other control flow

Reply:
wait hold up. If this isn't going on the guide, then we don't really have to be strict about this stuff. Do we?

Reply:
if this isn't going on the guide why is it in a pr on usaco-guide

Reply:
[Perhaps for this reason](https://github.com/cpinitiative/usaco-guide/pull/1959#discussion_r672660506)

Reply:
ooooooohhhh ok
uh still it's good to like make the code look good i mean
but yeah i won't lose my crap about this

Reply:
> uh still it's good to like make the code look good i mean
tbf there are a lot of solutions with shittier code than this so like i think we're fine idk.

Reply:
👀 😞 that's not a reason to not format your code nicely...